estela-web: Automatically try to add arguments, env vars, and tags written in creation forms

raymond1242 opened the pull request:
# Description

estela-web: Automatically try to add arguments, env vars, and tags written in creation forms

# Issue

[* _Github Issue ID_.](https://tasks.bitmaker.dev/issues/3719)

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] My code follows the style guidelines of this project.
- [ ] I have made corresponding changes to the [documentation](https://github.com/bitmakerla/estela/tree/main/docs).
- [ ] New and existing tests pass locally with my changes.
- [ ] If this change is a core feature, I have added thorough tests.
- [ ] If this change affects or depends on the behavior of other _estela_ repositories, I have created pull requests with the relevant changes in the affected repositories. Please, refer to our [official documentation](https://estela.bitmaker.la/docs/).
- [ ] I understand that my pull request may be closed if it becomes obvious or I did not perform all of the steps above.

mgonnav reviewed (changes requested):
@raymond1242 I have added a change refactoring some of the code. We still need to add this functionality to the `clone` modals.

Also, ensure that we verify both cases correctly when the name is present but the value isn't and vice-versa. I fixed it for the create modals because if I had an argument/env var `name:` it would raise an error but `:value` would create the job without verifying this is not a valid argument/env var.

raymond1242 commented:
> @raymond1242 I have added a change refactoring some of the code. We still need to add this functionality to the `clone` modals.
> 
> Also, ensure that we verify both cases correctly when the name is present but the value isn't and vice-versa. I fixed it for the create modals because if I had an argument/env var `name:` it would raise an error but `:value` would create the job without verifying this is not a valid argument/env var.

Done @mgonnav
